Fix OpenXR build failure when glTF module is disabled

Mrfanta-stick opened the pull request:
Title: Fix OpenXR build failure when glTF module is disabled

Description: When compiling with module_gltf_enabled=no, the OpenXR module fails to link because it assumes GLTFDocument is always available. Additionally, the test suite attempts to run GLTF tests even when the module is disabled.

This PR adds #ifdef MODULE_GLTF_ENABLED guards to OpenXRRenderModelExtension and test_main.cpp to properly respect the build option.

Fixes #113427

Mrfanta-stick commented:
Thanks for the feedback @dsnopek!

Apologies for the delay in getting this updated—I had some personal matters to attend to over the last few days.

As a first-time contributor, I really appreciate the guidance on the architectural approach here. It makes total sense to disable the feature entirely rather than patching the internals.

I've pushed the changes to fully exclude the OpenXRRenderModel extension (including its registration in register_types.cpp) when MODULE_GLTF_ENABLED is false. I also added guards to test_main.cpp to silence the GLTF tests in this build configuration.

AThousandShips commented on modules/openxr/scene/openxr_render_model_manager.cpp:
```diff
@@ -282,3 +285,4 @@ void OpenXRRenderModelManager::set_make_local_to_pose(const String &p_action) {
 String OpenXRRenderModelManager::get_make_local_to_pose() const {
 	return make_local_to_pose;
 }
+#endif
```
```suggestion
#endif // MODULE_GLTF_ENABLED
```

AThousandShips commented on modules/openxr/scene/openxr_render_model.h:
```diff
@@ -33,7 +33,7 @@
 
 #include <openxr/openxr.h>
-
+#ifdef MODULE_GLTF_ENABLED
```
```suggestion

#ifdef MODULE_GLTF_ENABLED
```

AThousandShips commented on modules/openxr/scene/openxr_render_model.h:
```diff
@@ -58,3 +58,4 @@ class OpenXRRenderModel : public Node3D {
 
 	String get_top_level_path() const;
 };
+#endif
```
```suggestion
#endif // MODULE_GLTF_ENABLED

```

AThousandShips commented on modules/openxr/scene/openxr_render_model.cpp:
```diff
@@ -166,3 +167,4 @@ void OpenXRRenderModel::set_render_model(RID p_render_model) {
 		_load_render_model_scene();
 	}
 }
+#endif
```
```suggestion
#endif // MODULE_GLTF_ENABLED
```

Mrfanta-stick commented:
@AThousandShips Thanks so much for the real-time review! I've tried and pushed fixes for the whitespace/comment style issues and corrected the include placement.

Regarding the includes: I was being overly aggressive with the #ifdef wrapping because I wanted to ensure strict isolation of the module dependencies. I didn't realize the main header needed to remain exposed to satisfy the self-contained header rule.

Appreciate your patience—I'm still getting the hang of the C++ / Godot contribution workflow, so the specific feedback is really helpful!

AThousandShips commented:
The whitespace changes are still left in the file, and the final lines are missing some line endings

AThousandShips commented:
Please set up [pre-commit locally](https://contributing.godotengine.org/en/latest/engine/guidelines/code_style.html#using-clang-format-locally) to make sure the files are properly formatted

Mrfanta-stick commented:
@AThousandShips Thanks a lot for the suggestion! Your guidance is much valued and appreciated as I was struggling with the file formatting manually. I set up the environment for the pre-commit, ran the checks (they passed locally), and pushed the fixes. Hopefully, it'll be green now.

AThousandShips commented:
There are some minor changes that should probably be removed but I can help you with that if you wish when this has been evaluated otherwise

dsnopek commented on modules/openxr/extensions/openxr_render_model_extension.cpp:
```diff
@@ -476,7 +477,6 @@ Node3D *OpenXRRenderModelExtension::render_model_new_scene_instance(RID p_render
 		// We never loaded it (don't spam errors here).
 		return nullptr;
 	}
-
```
This shouldn't be removing random lines

dsnopek commented on modules/openxr/extensions/openxr_render_model_extension.cpp:
```diff
@@ -728,7 +728,6 @@ Ref<OpenXRRenderModelData> OpenXRRenderModelExtension::_load_asset(XrRenderModel
 
 	Ref<OpenXRRenderModelData> render_model_data;
 	render_model_data.instantiate();
-
```
Here too

dsnopek commented on modules/openxr/scene/openxr_render_model.h:
```diff
@@ -34,6 +34,9 @@
 
+#include "modules/modules_enabled.gen.h"
+
+#ifdef MODULE_GLTF_ENABLED
```
Same as earlier, this should be moved up to be under the `#pragma once`

dsnopek commented on modules/openxr/scene/openxr_render_model_manager.h:
```diff
@@ -30,14 +30,19 @@
 
 #include <openxr/openxr.h>
 
+#ifdef MODULE_GLTF_ENABLED
```
Here too

dsnopek reviewed (commented):
Thanks!

At a high-level, this is the right overall approach, but there's a bunch of little issues: some unrelated changes, the position of `#ifdef`'s to remove as much as possible, etc. I've called out everything I noticed below.

Also, this is up to 14 commits - before it can be merged, you'll need to squash this down to 1 commit per Godot's Git workflow - see:

https://contributing.godotengine.org/en/latest/organization/pull_requests/creating_pull_requests.html#the-interactive-rebase

Mrfanta-stick committed:
Fix OpenXR build failure when GLTF module is disabled

Wrap OpenXR render model classes with MODULE_GLTF_ENABLED guards since
they depend on the GLTF module. Include modules_enabled.gen.h
unconditionally to ensure proper module detection across all build
configurations.

dsnopek reviewed (approved):
Thanks! This looks good to me now :-)

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
